src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -267,24 +249,32 @@ public void commandFailed(final CommandFailedEvent event) {
   }
 
   private String getMBeanName(final String mBean) {
-    return "com.mongodb:type=MongoDBKafkaConnector,name=" + mBean + id;
+    String id = MBeanServerUtils.taskIdFromCurrentThread();
+    return "com.mongodb.kafka.connect:type=source-task-metrics,task=" + mBean + "-" + id;
   }
 
   @Override
   public List<SourceRecord> poll() {
+    if (!isCopying.get()) {
+      currentStatistics = streamStatistics;
+    }
     if (lastTaskInvocation != null) {
-      currentStatistics.timeSpentOutsidePollTask(lastTaskInvocation);
+      currentStatistics
+          .getBetweenTaskInvocations()
+          .sample(lastTaskInvocation.getElapsedTime(TimeUnit.MILLISECONDS));
     }
-    Timer taskTime = currentStatistics.taskInvoked();
+    Timer taskTime = Timer.start();
     List<SourceRecord> sourceRecords = pollInternal();
     if (sourceRecords != null) {
-      currentStatistics.returnedRecords(sourceRecords.size());
+      currentStatistics.getRecordsReturned().sample(sourceRecords.size());
     }
-    if (!isCopying.get()) {
-      currentStatistics = streamStatistics;
+    currentStatistics.getTaskInvocations().sample(taskTime.getElapsedTime(TimeUnit.MILLISECONDS));
+    lastTaskInvocation = Timer.start();
+
+    if (LOGGER.isDebugEnabled()) {
```

Comment:
I think it would be better to only print the current statistic.  When we're in the copy state, stream statistics are empty.  And once we're in stream state, copy statistics no long change (I think).

Reply:
Why would it be useful to separate copy and stream in JMX, but not in logging?

Reply:
Separating is still useful.  I think the analogy would be if we were unregister the copy mbean once the copying phase is over.  And maybe that's a good idea too.

Reply:
I changed it to log only the current.

I don't think we should unregister the copy mbean, because it might be sampled, and then accrue some counts, and then be unregistered before it is sampled again. (This is incidentally not an issue with logging, since no counts are accrued between logging and the switch from copy to stream.)